modules/web/src/app/node-data/component.ts

```diff
@@ -548,8 +548,8 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
     const selectedOperatingSystem = this.form.get(Controls.OperatingSystem).value;
     const dcOSP = this._datacenterSpec?.spec.operatingSystemProfiles?.[selectedOperatingSystem];
 
-    if (this.selectedOperatingSystemProfile) {
-      ospValue = this.selectedOperatingSystemProfile;
+    if (this._nodeDataService.nodeData.operatingSystemProfile) {
+      ospValue = this._nodeDataService.nodeData.operatingSystemProfile;
```

Comment:
We reset the selected OSP value when user selects a different OS. Can you please double check that this still works? Like the selected OSP value is reset when OS changes.

Ref:
``` typescript
merge(this.form.get(Controls.OperatingSystem).valueChanges)
      .pipe(takeUntil(this._unsubscribe))
      .subscribe(_ => {
        // We don't want to retain the existing value for OSP in the edit view since user explicitly
        // changed the selected operating system.
        this.selectedOperatingSystemProfile = null;
        this.supportedOperatingSystemProfiles = this.getSupportedOperatingSystemProfiles();
        this.setDefaultOperatingSystemProfiles();
      });